src/rpc/rawtransaction.cpp

```diff
@@ -363,7 +366,106 @@ static UniValue verifytxoutproof(const JSONRPCRequest& request)
     return res;
 }
 
-CMutableTransaction ConstructTransaction(const UniValue& inputs_in, const UniValue& outputs_in, const UniValue& locktime, const UniValue& rbf, const UniValue& assets_in, std::vector<CPubKey>* output_pubkeys_out)
+template<typename T_tx_ref, typename T_merkle_block>
```

Comment:
Suggested comment:

`Modifies an existing transaction input in-place to be a valid peg-in input, and inserts the witness if deemed valid.`

Reply:
This function should probably abort if the input already is marked, or otherwise has witness/scriptSig data, yes?

Reply:
I've added the comment and an abort if there is a scriptSig or witness.